internal/services/servicebus/servicebus_namespace_resource.go

```diff
@@ -507,20 +608,34 @@ func expandServiceBusNamespaceEncryption(input []interface{}) *namespaces.Encryp
 	v := input[0].(map[string]interface{})
 	keyId, _ := keyVaultParse.ParseOptionallyVersionedNestedItemID(v["key_vault_key_id"].(string))
 	keySource := namespaces.KeySourceMicrosoftPointKeyVault
-	return &namespaces.Encryption{
-		KeyVaultProperties: &[]namespaces.KeyVaultProperties{
+
+	encryption := namespaces.Encryption{
+		KeySource:                       &keySource,
+		RequireInfrastructureEncryption: pointer.To(v["infrastructure_encryption_enabled"].(bool)),
+	}
+
+	if v["identity_id"].(string) == "" {
+		encryption.KeyVaultProperties = &[]namespaces.KeyVaultProperties{
+			{
+				KeyName:     pointer.To(keyId.Name),
+				KeyVersion:  pointer.To(keyId.Version),
+				KeyVaultUri: pointer.To(keyId.KeyVaultBaseUrl),
+			},
+		}
```

Comment:
If we allow `identity_id` to be `Optional`, can users actually create a servicebus namespace with CMK using `SystemAssigned` identity on first go?

Reply:
they can't, so I've reverted this to how it was before 👍